opentracing-spring-web-starter/src/main/java/io/opentracing/contrib/spring/web/starter/SkipPatternAutoConfiguration.java

```diff
@@ -0,0 +1,160 @@
+import org.springframework.boot.actuate.endpoint.web.ExposableWebEndpoint;
+import org.springframework.boot.actuate.endpoint.web.PathMappedEndpoint;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.web.ServerProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.StringUtils;
+
+import io.opentracing.contrib.spring.web.webfilter.SkipPattern;
+
+/**
+ * @author Gilles Robert
+ */
+@Configuration
+@ConditionalOnProperty(name = "opentracing.spring.web.enabled", havingValue = "true", matchIfMissing = true)
+public class SkipPatternAutoConfiguration {
+
+  @Autowired(required = false)
+  private List<SkipPattern> patterns = new ArrayList<>();
+
+  @Bean(name = "skipPattern")
+  public Pattern skipPattern() {
+    return Pattern.compile(this.patterns
+        .stream()
+        .map(SkipPattern::pattern)
+        .filter(Optional::isPresent).map(Optional::get)
+        .map(Pattern::pattern)
+        .collect(Collectors.joining("|")));
+  }
+
+  @Configuration
+  @ConditionalOnClass(ManagementServerProperties.class)
+  @ConditionalOnProperty(value = "opentracing.spring.web.ignoreAutoConfiguredSkipPatterns", havingValue = "false", matchIfMissing = true)
+  protected static class ManagementSkipPatternProviderConfig {
+
+    static Optional<Pattern> getPatternForManagementServerProperties(
+        ManagementServerProperties managementServerProperties) {
+      String contextPath = managementServerProperties.getServlet().getContextPath();
+      if (StringUtils.hasText(contextPath)) {
+        return Optional.of(Pattern.compile(contextPath + ".*"));
+      }
+      return Optional.empty();
+    }
+
+    @Bean
+    @ConditionalOnBean(ManagementServerProperties.class)
+    public SkipPattern skipPatternForManagementServerProperties(
+        final ManagementServerProperties managementServerProperties) {
+      return () -> getPatternForManagementServerProperties(managementServerProperties);
+    }
+  }
+
+  @Configuration
+  @ConditionalOnClass( {ServerProperties.class, EndpointsSupplier.class, ExposableWebEndpoint.class})
+  @ConditionalOnBean(ServerProperties.class)
+  @ConditionalOnProperty(value = "opentracing.spring.web.ignoreAutoConfiguredSkipPatterns", havingValue = "false", matchIfMissing = true)
+  protected static class ActuatorSkipPatternProviderConfig {
+
+    static Optional<Pattern> getEndpointsPatterns(String contextPath,
+                                                  WebEndpointProperties webEndpointProperties,
+                                                  EndpointsSupplier<ExposableWebEndpoint> endpointsSupplier) {
+      Collection<ExposableWebEndpoint> endpoints = endpointsSupplier.getEndpoints();
```

Comment:
@ask4gilles see https://github.com/opentracing-contrib/java-spring-cloud/issues/219. When I update this library in sping cloud it throws an exception in zuul module:

> Error creating bean with name 'org.springframework.cloud.client.CommonsClientAutoConfiguration$ActuatorConfiguration': Unsatisfied dependency expressed through field 'hasFeatures'